diff --git a/sshd.c b/sshd.c
index 2a2d149..625be73 100644
--- a/sshd.c
+++ b/sshd.c
@@ -1486,8 +1486,8 @@ check_ip_options(struct ssh *ssh)
 #ifdef IP_OPTIONS
 	int sock_in = ssh_packet_get_connection_in(ssh);
 	struct sockaddr_storage from;
-	socklen_t option_size, i, fromlen = sizeof(from);
 	u_char opts[200];
+	socklen_t i, option_size = sizeof(opts), fromlen = sizeof(from);
 	char text[sizeof(opts) * 3 + 1];
 
 	memset(&from, 0, sizeof(from));

